economy/models.py

```diff
@@ -117,19 +117,16 @@ def terminate_active_session(cls):
             active_session.save()
 
     @property
-    def total_purchases(self) -> int:
-        return self.purchases.count()
+    def total_product_orders(self) -> int:
+        return self.product_orders.count()
 
     @property
-    def total_amount(self) -> int:
-        total_amount = 0
-        for purchase in self.purchases.all():
-            total_amount += purchase.total_amount
-
-        return total_amount
+    def total_revenue(self) -> int:
+        purchase_sums = [amount.cost for amount in self.product_orders.all()]
+        return sum(purchase_sums)
```

Comment:
Could probably be done using DB field annotation, but couldn't really make out how to go about it